File: packages/domain/src/entities/history.ts
export interface HistoryEntry {
  id: string;
  request_id: string | null;
  workspace_id: string;
  method: string;
  url: string;
  request_snapshot: string;
  response_status: number | null;
  response_headers: string | null;
  response_body: string | null;
  response_size: number | null;
  duration_ms: number | null;
  error: string | null;
  executed_at: string;
}

export interface HistoryQuery {
  workspace_id: string;
  limit?: number;
  offset?: number;
}

File: apps/desktop/src-tauri/src/models/history.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HistoryEntry {
    pub id: String,
    pub request_id: Option<String>,
    pub workspace_id: String,
    pub method: String,
    pub url: String,
    pub request_snapshot: String,
    pub response_status: Option<i32>,
    pub response_headers: Option<String>,
    pub response_body: Option<String>,
    pub response_size: Option<i64>,
    pub duration_ms: Option<i64>,
    pub error: Option<String>,
    pub executed_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HistoryQuery {
    pub workspace_id: String,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}
